src/librustdoc/html/layout.rs

```diff
@@ -36,7 +36,8 @@ pub fn render<T: fmt::Display, S: fmt::Display>(
     -> io::Result<()>
 {
     write!(dst,
-r##"<!DOCTYPE html>
+r##"<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
+     "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
```

Comment:
yeah, this changes from HTML5 to XHTML 1.0, which we shouldn't do. this is the correct doctype